### PR TITLE
MTA-2696: Documentation for multilanguage support

### PR DESCRIPTION
**Pull request by vsodhi**
### JIRA

* [MTA-2696](https://issues.redhat.com/browse/MTA-2696)

Support for .NET language added

### Preview

https://vsodhi.github.io/previews/MTA/MTA-2696.html#_rulesets

### CODE REVIEW

**Review by rromannissen — CHANGES_REQUESTED** [review]
I don't see any capabilities for the .NET provider being added in 2.2.3, nor any explanation of the .NET provider itself as the ones we have for the builtin, java and go providers right after the change that was included. This is not only not complete, it's not even started.

File: docs/topics/yaml-rule-structure-syntax.adoc [hunk 1]
```diff
@@ -246,6 +246,7 @@ The analyzer currently supports the following `provider` conditions:
 * `builtin`
 * `java`
 * `go`
+* `.NET`
```
Comment (dymurray) [inline docs/topics/yaml-rule-structure-syntax.adoc:249]: this is `dotnet`

Comment (vsodhi) [reply docs/topics/yaml-rule-structure-syntax.adoc]: fixed.

**Comment by anarnold97** [comment]
@midays - please can you review. thanks

**Review by dymurray — APPROVED** [review]

File: docs/topics/yaml-rule-structure-syntax.adoc [hunk 2]
```diff
@@ -230,6 +230,9 @@ when:
 
 ==== Provider conditions
 
+A provider detects the programming languages, frameworks, and tools used to build an application and creates default rulesets for each supported provider using the Language Server Protocol (LSP) accordingly.
```
[the remark this answers is not in the document]
Comment (vsodhi) [reply docs/topics/yaml-rule-structure-syntax.adoc]: fixed.

**Review by midays — APPROVED** [review]
LGTM

File: docs/topics/yaml-rule-structure-syntax.adoc [hunk 3]
```diff
@@ -246,6 +249,29 @@ The analyzer currently supports the following `provider` conditions:
 * `builtin`
 * `java`
 * `go`
+* `dotnet`
++
+:FeatureName: dotnet provider
+include::../snippets/technology-preview-admonition.adoc[]
+
+[[tabl-Supported-providers-and-rule-conditions]]
```
Comment (anarnold97) [inline docs/topics/yaml-rule-structure-syntax.adoc:257]:
```suggestion
[[table-Supported-providers-and-rule-conditions]]
```
Pedantic... very pedantic

Comment (vsodhi) [reply docs/topics/yaml-rule-structure-syntax.adoc]: good catch. fixed.

**Pull request merged**